test/org/javarosa/test/utils/ResourcePathHelper.java

```diff
@@ -6,16 +6,34 @@
 
 public class ResourcePathHelper {
 
-    /** Makes a Path for a resource file */
+    /**
+     * Makes a Path for a resource file either in a directory corresponding to the test class’s package, or
+     * in the resources directory.
+     *
+     * @param filename the file name for which to create a path
+     * @return a Path for the resource file
+     */
     public static Path r(String filename) {
+        return r(filename, true);
+    }
+
+    /**
+     * Makes a Path for a resource file either in a directory corresponding to the test class’s package, or
+     * in the resources directory.
+     *
+     * @param filename the file name for which to create a path
+     * @param fallBack whether to “fall back” to the resources directory if the file is not in the
+     *                 class’s corresponding directory
+     * @return a Path for the resource file
+     */
+    public static Path r(String filename, boolean fallBack) {
```

Comment:
I think we should throw an error when a test asks for a file that isn't there. In fact, this implementation in masking an error on this PR: the file `towns-large.xml`is not present (I'm guessing you forgot to commit the file?), but the timing test still passes.

I like the fallback mechanism though, and I think we should have it as the default behavior, which would avoid some changes on the test classes. I've tried this implementation, which works:

```java
public static Path r(String filename) {
    String resourceFileParentPath = inferResourceFileParentPath();
    Path resourceFilePath = Paths.get("resources", resourceFileParentPath, filename);

    if (Files.exists(resourceFilePath))
        return resourceFilePath;

    Path resources = Paths.get("resources", filename);
    if (Files.exists(resources))
        return resources;

    throw new RuntimeException("File " + filename + " not found");
}
```

Reply:
Now I think I'd like to change the implementation to recursively search for the file with `Files.walk`, but I'm not super crazy about it either.

Reply:
`towns-large.xml` is dynamically generated, hence it might not exist when `r` is called.

Reply:
Ugh, I didn't realize that... 

In any case, what do you think about making it the default behavior and avoid having two `parse` methods?

This implementation works on all the test scenarios without polluting unknown paths:
```java
public static Path r(String filename) {
    String resourceFileParentPath = inferResourceFileParentPath();
    Path resourceFilePath = Paths.get("resources", resourceFileParentPath, filename);

    if (Files.exists(resourceFilePath))
        return resourceFilePath;

    Path resources = Paths.get("resources", filename);
    if (Files.exists(resources))
        return resources;

    // File doesn't exist. Returning a temp file
    String name = filename.substring(0, filename.lastIndexOf("."));
    String extension = filename.substring(filename.lastIndexOf("."));
    try {
        return Files.createTempFile(name + "-", extension);
    } catch (IOException e) {
        throw new UncheckedIOException(e);
    }
}
```

This code would create a `/tmp/towns-large-8296528288103120809.xml`, for example.